components/brave_rewards/browser/test/rewards_p3a_browsertest.cc

```diff
@@ -140,6 +175,86 @@ IN_PROC_BROWSER_TEST_F(RewardsP3ABrowserTest, RewardsDisabled) {
+  // process, which interferes with other tests running in parallel.
+  // Instead rely on the fact that the EnabledDuration P3A measurement
+  // is made by the rewards service preference observer.
+  prefs->SetBoolean("brave.brave_ads.enabled", false);
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kHours, 1);
+
+  // Mock turning rewards back on.
+  prefs->SetBoolean("brave.brave_ads.enabled", true);
+  // Adjust the stored timestamp to measure a longer duration.
+  auto earlier = base::Time::Now() - base::Minutes(90);
+  LOG(INFO) << "Backdating timestamp to " << earlier;
+  prefs->SetTime(brave_rewards::prefs::kEnabledTimestamp, earlier);
+
+  // Mock turning rewards off.
+  prefs->SetBoolean("brave.brave_ads.enabled", false);
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kHours, 2);
+
+  // Mock turning rewards back on.
+  prefs->SetBoolean("brave.brave_ads.enabled", true);
+  auto yesterday = base::Time::Now() - base::Days(1);
+  LOG(INFO) << "Backdating timestamp to " << yesterday;
+  prefs->SetTime(brave_rewards::prefs::kEnabledTimestamp, yesterday);
+
+  // Mock turning rewards off.
+  prefs->SetBoolean("brave.brave_ads.enabled", false);
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kDays, 1);
+
+  // Mock turning rewards on for more than a week.
+  prefs->SetBoolean("brave.brave_ads.enabled", true);
+  auto last_week = base::Time::Now() - base::Days(12);
+  LOG(INFO) << "Backdating timestamp to " << last_week;
+  prefs->SetTime(brave_rewards::prefs::kEnabledTimestamp, last_week);
+
+  // Mock turning rewards off.
+  prefs->SetBoolean("brave.brave_ads.enabled", false);
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kWeeks, 1);
+
+  // Mock turning rewards on for more than a month.
+  prefs->SetBoolean("brave.brave_ads.enabled", true);
+  auto last_month = base::Time::Now() - base::Days(40);
+  LOG(INFO) << "Backdating timestamp to " << last_month;
+  prefs->SetTime(brave_rewards::prefs::kEnabledTimestamp, last_month);
+
+  // Mock turning rewards off.
+  prefs->SetBoolean("brave.brave_ads.enabled", false);
+  histogram_tester_->ExpectBucketCount("Brave.Rewards.EnabledDuration",
+                                       RewardsEnabledDuration::kMonths, 1);
+
+  // Mock turning rewards on for our longest measured value.
+  prefs->SetBoolean("brave.brave_ads.enabled", true);
+  auto long_ago = base::Time::Now() - base::Days(128);
+  LOG(INFO) << "Backdating timestamp to " << long_ago;
+  prefs->SetTime(brave_rewards::prefs::kEnabledTimestamp, long_ago);
+
+  // Mock turning rewards off.
+  prefs->SetBoolean("brave.brave_ads.enabled", false);
```

Comment:
why not `prefs::kEnabled` everywhere?

Reply:
Originally there was only one reference, and I had trouble finding the correct header. Fixed in 55ec9c0687bfb68742af6498bc76f0243aa06b1e.